src/Entity/Item.php

```diff
@@ -17,67 +17,57 @@ class Item implements \JsonSerializable
     #[ORM\Column]
     private ?int $valor;
 
-    #[ORM\OneToOne(cascade: ['persist', 'remove'])]
+    #[ORM\ManyToOne(targetEntity: Produto::class)]
+    #[ORM\JoinColumn(nullable: false)]
     private ?Produto $produto;
 
-    #[ORM\ManyToOne(inversedBy: 'items')]
+    #[ORM\ManyToOne(targetEntity: Carrinho::class , inversedBy: 'items')]
+    #[ORM\JoinColumn(nullable: false)]
     private ?Carrinho $carrinho;
 
-    public function getId(): ?int
+    #[ORM\Column]
+    private int $quantidade;
+
+    public function __construct(
+        Carrinho    $carrinho,
+        Produto $produto,
+        int $valor,
+        int $quantidade
+    )
     {
```

Comment:
legal começar a observar como fica a identação e tals...

```suggestion
    public function __construct(
        Carrinho $carrinho,
        Produto $produto,
        int $valor,
        int $quantidade
    ) {
```